Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  ObjectiveCDictionaryExtension.swift
+//  plank
+//
+//  Created by Martin Matejovic on 18/05/2017.
+//
+//
+
+import Foundation
+extension ObjCModelRenderer {
+    func renderGenerateDictionary() -> ObjCIR.Method {
+        let props = self.properties.map { (param, schema) -> String in
+            ObjCIR.ifStmt("_"+"\(self.dirtyPropertiesIVarName).\(dirtyPropertyOption(propertyName: param, className: self.className))") {
+                return [renderAddObjectStatement(param, schema)]
+            }
+            }.joined(separator: "\n")
+        
+        return ObjCIR.method("- (NSDictionary *)dictionaryRepresentation") {[
+            "NSMutableDictionary *dict = [[NSMutableDictionary alloc]initWithCapacity:\(self.properties.count)];",
+            props,
+            "return dict;"
+            ]}
+    }
+    
+}
+
+
+extension ObjCFileRenderer {
+    fileprivate func renderAddObjectStatement(_ param: String, _ schema: Schema) -> String {
+        let propIVarName = "_\(param.snakeCaseToPropertyName())"
+        switch schema {
+        case .boolean:
+            return "[dict setObject: [NSNumber numberWithBool:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+
+        case .float:
+            return "[dict setObject: [NSNumber numberWithFloat:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .integer:
+            return "[dict setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .object(_):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
```

Comment:
This is another model type so we will want to set the `dictionaryRepresentation` of that object rather than just storing it in a dictionary.